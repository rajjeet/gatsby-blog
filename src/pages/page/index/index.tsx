import React from 'react';
import { graphql } from 'gatsby';
import Layout from '../../../components/layout';
import PostListing from '../../../components/post-listing';
import Seo from '../../../components/seo';
import ProjectListing from '../../../components/project-listing';
import { TPostEdge, TImage } from '../../../templates/category-page/types';

export type TProjectEdge = {
  node: {
    frontmatter: {
      title: string;
      description: string;
      thumbnail: TImage;
      techStackTags: {label: string; type: string}[];
      links: {label: string; value: string}[];
    };
    fields: {
      slug: string;
    };
    body: string;
  };
};

export type TProps = {
  data: {
    posts: {
      edges: TPostEdge[];
    };
    projects: {
      edges: TProjectEdge[];
    };
  };
};

const IndexPage: React.FC<TProps> = ({ data }) => (
  <Layout>
    <Seo />
    <ProjectListing
      projects={data.projects.edges}
      heading="Side Projects"
    />
    <PostListing
      posts={data.posts.edges}
      heading="Latest Posts"
    />
  </Layout>
);

export default IndexPage;

export const query = graphql`
    {
        posts: allMdx(sort: {fields: [frontmatter___date], order: DESC}, filter: {frontmatter: {draft: {ne: true}}, fields: {contentType: {eq: "post"}}}, limit: 3) {
            edges {
                node {
                    ...PostListingMarkdownFragment
                }
            }
        }
        projects: allMdx(limit: 2, sort: {fields: frontmatter___date, order: DESC}, filter: {fields: {contentType: {eq: "project"}}}) {
            edges {
                node {
                    ...ProjectListingMarkdownFragment
                }
            }
        }
    }
`;
