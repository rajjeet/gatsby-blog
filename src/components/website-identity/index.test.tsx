import React from 'react';
import { render } from '@testing-library/react';
import { WebsiteIdentity } from './index';

describe('<WebsiteIdentity />', () => {
  it('should render', () => {
    const { asFragment } = render(<WebsiteIdentity />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have Ortmesh title', () => {
    const { getByText } = render(<WebsiteIdentity />);
    expect(getByText('Ortmesh')).toBeDefined();
  });

  it('should have an image with website logo', () => {
    const { getByAltText } = render(<WebsiteIdentity />);
    expect(getByAltText(/ortmesh/i)).toBeDefined();
  });

  it('should have the website caption', () => {
    const { getByText } = render(<WebsiteIdentity />);
    expect(getByText('Write code that matters')).toBeDefined();
  });
});
