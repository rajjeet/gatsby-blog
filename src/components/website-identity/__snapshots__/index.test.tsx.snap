// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WebsiteIdentity /> should render 1`] = `
<DocumentFragment>
  .c2 {
  width: 3rem;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration-color: inherit;
  text-decoration-color: inherit;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 190px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-size: 1rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  font-size: 0.8rem;
  margin: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  margin: 0;
}

<a
    aria-label="See Home"
    class="c0"
    href="/"
  >
    <div
      class="c1"
    >
      <div
        class="c2 gatsby-image-wrapper"
        style="position: relative; overflow: hidden;"
      >
        <div
          style="width: 100%; padding-bottom: 100%;"
        />
        <picture>
          <source
            sizes=""
            srcset=""
          />
          <img
            alt="Ortmesh logo"
            loading="lazy"
            sizes=""
            src="https://via.placeholder.com/250x180"
            srcset=""
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="https://via.placeholder.com/250x180" alt="Ortmesh logo" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Ortmesh
        </h2>
        <p
          class="c5"
        >
           Write code that matters 
        </p>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
